runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/ConvexHull.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2018, James Swindle <wilingua@gmail.com>
+ * All rights reserved.
+ * 
+ * Algorithm credit goes to WikiBooks
+ * https://en.wikibooks.org/wiki/Algorithm_Implementation/Geometry/Convex_hull/Monotone_chain#Java
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.npchighlight;
+
+import java.awt.Polygon;
+import java.awt.geom.Area;
+import java.awt.geom.PathIterator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import net.runelite.api.Point;
+
+public class ConvexHull
+{
+
+	public static long cross(Point O, Point A, Point B)
+	{
+		return (A.getX() - O.getX()) * (B.getY() - O.getY()) - (A.getY() - O.getY()) * (B.getX() - O.getX());
+	}
+
+	public static Point[] convertArea(Area area)
+	{
+		PathIterator iterator = area.getPathIterator(null);
+		float[] floats = new float[6];
+		List<Point> points = new ArrayList<>();
+		while (!iterator.isDone())
+		{
+			int type = iterator.currentSegment(floats);
+			int x = (int) floats[0];
+			int y = (int) floats[1];
+			if (type != PathIterator.SEG_CLOSE)
+			{
+				points.add(new Point(x, y));
+			}
+			iterator.next();
+		}
+
+		return points.toArray(new Point[0]);
+	}
+
+	public static Polygon convertPoints(Point[] points)
+	{
+		Polygon poly = new Polygon();
+		for (int i = 0; i < points.length; i++)
+		{
+			poly.addPoint(points[i].getX(), points[i].getY());
+		}
+
+		return poly;
+	}
+
+	public static Polygon convexHull(Area area)
```

Comment:
There is already a Jarvis march implementation in the api. Is that not sufficient for this?

Reply:
I wasn't unaware of that, I've removed my implementation